New API landing page (#7460)
DataDog/documentation: content/en/api/_index.md

{{< h2 >}}API Reference{{< /h2 >}}

Use the Datadog HTTP API to programmatically access the Datadog platform. The Datadog API uses resource-oriented URLs, uses status codes to indicate the success or failure of requests and returns JSON from all requests.

**To get started on the Datadog HTTP API, use the following flow to determine which endpoints you need and checkout our [Datadog Postman collection][1].**

> {{< img src="api/api-flow.png" alt="Flow through Datadog" responsive="true" style="width:100%;">}}

Reviewer: ```suggestion
```

**Note**: cURL code examples assume usage of BASH and GNU coreutils. On macOS you can install coreutils via the [Homebrew package manager][2]: `brew install coreutils`

> 

Reviewer: ```suggestion

{{< img src="api/api-flow.png" alt="Flow through Datadog" responsive="true" style="width:100%;">}}
```

- The [events][13] endpoints allow you to post and fetch events to and from the [Datadog event stream][14].
- Use the [Synthetics][15] endpoints to create, start, stop and see [Synthetics tests][16] results.
- Use the [Tracing Agent API][17] to send traces to your Datadog Agent, which then forwards them to Datadog.

{{< h2 >}}Visualize your data{{< /h2 >}}

Once you are sending data to Datadog, you can use the API to programatically build data visualizations:

- Build [Dashboards][18] and view [Dashboard Lists][19]
- Manage [host tags][20]
- Create [Embeddable Graphs][21]
- Take a [graph snapshot][22]
- [Service Dependencies][23] - see a list of your APM services and their dependencies
- Create [Monitors][24]
- [Service Checks][25] - post check statuses for use with monitors
- Create and manage [Logs][26], [Logs Indexes][27], and [Logs Pipelines][28]
- Get [Host][20] information for your organization
- Create and manage [Service Level Objectives][29]
- Generate [Security Monitoring][30] signals

{{< h2 >}}Manage your account{{< /h2 >}}

You can also use the Datadog API to programatically manage your account:

- Manage [Users][31]

> - Mange [Roles][32]

Reviewer: ```suggestion
- Manage [Roles][32]
```

- Manage your [Organization][33]
- Veirfy API and app keys with the [Authentication][34] endpoint
- Grant specific logs access with the [Logs Restriction Queries][35]

> - Manage existing keys with[Key Management][36]

Reviewer: ```suggestion
- Manage existing keys with [Key Management][36]
```

> - Get get hourly, daily, and monthly usage across multiple facets of Datadog with the[Usage Metering][37] endpoints

Reviewer: ```suggestion
- Get get hourly, daily, and monthly usage across multiple facets of Datadog with the [Usage Metering][37] endpoints
```

- See the list of IP prefixes belonging to Datadog with[IP Ranges][38]

{{< h2 >}}Rate limiting{{< /h2 >}}

Some of the API endpoints are rate limited. Once you exceed a certain number of requests in a certain time period we return an error.

For rate limited API endpoints we return headers so you can know how close you are to your limit. If you exceed your limit, review these headers to determine when you are able to try again.

Rate limits can be increased from defaults by [contacting the Datadog support team][39].

Regarding API rate limit policy:

- Datadog **does not rate limit** on data point/metric submission (see [metrics section][11] for more info on how metric submission rate is handled). Limits encounter is dependent on the quantity of [custom metrics][40] based on your agreement.
- The rate limit for metric **retrieval** is `100` per hour per organization.
- The rate limit for event submission is `1000` per aggregate per day per organization. An aggregate is a group of similar events.

> - The rate limit for the [query_batch API][41] call is `600` per hour per organization. This can be extended on demand.

Reviewer: ```suggestion
- The rate limit for the [Query a Timeseries API][41] call is `600` per hour per organization. This can be extended on demand.
```

- The rate limit for the [Log Query API][42] call is `300` per hour per organization. This can be extended on demand.

> - The rate limit for the [graph_snapshot API][22] call is `60` per hour per organization. This can be extended on demand.

Reviewer: ```suggestion
- The rate limit for the [Graph a Snapshot API][22] call is `60` per hour per organization. This can be extended on demand.
```
